src/sentry/relay/config/metric_extraction.py

```diff
@@ -0,0 +1,130 @@
+from dataclasses import dataclass
+from typing import Any, List, Literal, Sequence, TypedDict, Union
+
+from sentry.api.endpoints.project_transaction_threshold import DEFAULT_THRESHOLD
+from sentry.models import (
+    Project,
+    ProjectTransactionThreshold,
+    ProjectTransactionThresholdOverride,
+    TransactionMetric,
+)
+
+
+class RuleConditionInner(TypedDict):
+    op: Literal["eq", "gt"]
+    name: str
+    value: Any
+
+
+# mypy does not support recursive types. type definition is a very small subset
+# of the values relay actually accepts
+class RuleCondition(TypedDict):
+    op: Literal["and"]
+    inner: Sequence[RuleConditionInner]
+
+
+class MetricConditionalTaggingRule(TypedDict):
+    condition: RuleCondition
+    targetMetrics: Sequence[str]
+    targetTag: str
+    tagValue: str
+
+
+_TRANSACTION_METRICS_TO_RULE_FIELD = {
+    TransactionMetric.LCP.value: "transaction.measurements.lcp",
+    TransactionMetric.DURATION.value: "transaction.duration",
+}
+
+_SATISFACTION_TARGET_METRICS = frozenset(
+    [
+        "s:transactions/user@none",
+        "d:transactions/duration@millisecond",
+    ]
+)
+
+_SATISFACTION_TARGET_TAG = "satisfaction"
+
+
+@dataclass
+class _DefaultThreshold:
+    metric: TransactionMetric
+    threshold: int
+
+
+_DEFAULT_THRESHOLD = _DefaultThreshold(
+    metric=TransactionMetric[DEFAULT_THRESHOLD["metric"].upper()].value,
+    threshold=int(DEFAULT_THRESHOLD["threshold"]),
+)
+
+
+def get_metric_conditional_tagging_rules(
+    project: Project,
+) -> Sequence[MetricConditionalTaggingRule]:
+    rules: List[MetricConditionalTaggingRule] = []
+
+    for threshold in project.projecttransactionthresholdoverride_set.all():
+        rules.extend(
+            _threshold_to_rules(
+                threshold,
+                [{"op": "eq", "name": "event.transaction", "value": threshold.transaction}],
+            )
+        )
+
+    try:
+        threshold = ProjectTransactionThreshold.objects.get(project=project)
```

Comment:
Just to double check: When there are more than one `ProjectTransactionThreshold`s, the exception raised is caught by the catch-all on [line 200](https://github.com/getsentry/sentry/pull/33722/files#diff-5e30755537fcfd5687d4399272065e49affcc12a4cf7a0bc377cff5234b8a08aR200), right?

Reply:
yup I changed that behavior, there's a unique constraint on `project_id` on this table